src/CMakeLists.txt

```diff
@@ -342,12 +367,30 @@ else() # Linux
     RUNTIME_OUTPUT_DIRECTORY_RELWITHDEBINFO "${SM_ROOT_DIR}"
   )
 
+  if(${WITH_CRASH_HANDLER})
+    sm_add_compile_definition("${SM_EXE_NAME}" CRASH_HANDLER)
+    if(LINUX)
+      # used only in the ArchHooks_Unix.cpp file
```

Comment:
If this is only used with the `ArchHooks_Unix.cpp` file, then wouldn't it make more sense to add these preprocessor definitions only to the individual source files instead?